### PR TITLE
My attempt at the airport challenge

### PR DESCRIPTION
**Pull request by ashtroboy79**
This is my attempt at the first weekend challenge.

### CODE REVIEW

**Comment by ashtroboy79** [comment]
I made it through all the user stories and have looked at possible edge cases. I would like a general review of my class structure, and how you would look at the edge cases. Also if you could look at my rspec tests and see how they could be improved

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  def takeoff_airplane(plane)
+    fail "plane not at airport" unless planes.include?(plane)
+    fail "weather is stormy" if stormy?
+    
+    plane.flying?(true)
+    planes.delete(plane)
+    puts "plane no longer at airport"
+  end
+
+  def stormy?
```
Comment (edpackard) [inline lib/airport.rb:31]: It is a bit confusing that you have a `stormy?` method in both the Airport and Weather class

File: lib/plane.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Plane
+
+  attr_reader :flying
+  def initialize
+    @flying = true
+  end
+  
+  def flying?(status = true)
```
[the remark this answers is not in the document]
Comment (ashtroboy79) [reply lib/plane.rb:8]: i shall change it to just a standard method. Is it because they have similar names? I shall have to work on making sure variables and methods don't share names.

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+require 'airport'
+
+describe Airport do
+  it 'lands a plane' do
+    expect(subject).to respond_to :land_airplane
+  end
+
+  it 'a plane can take-off' do
+    plane = Plane.new
```
[the remark this answers is not in the document]
Comment (ashtroboy79) [reply spec/airport_spec.rb:9]: I think this was me being scared/uncertain of how doubling would work, even after watching some videos, and doing some reading, but that is definitely a good idea.

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    allow(subject).to receive(:stormy?) { false }
+    subject.land_airplane(plane)
+    subject.takeoff_airplane(plane)
+    expect(subject.planes).not_to include(plane)
+  end
+
+  it 'prevents landing if airport is full' do
+    allow(subject).to receive(:stormy?) { false }
+    expect { (Airport::DEFAULT_CAPAPCITY + 1).times { subject.land_airplane Plane.new } }.to raise_error "airport full"
```
[the remark this answers is not in the document]
Comment (ashtroboy79) [reply spec/airport_spec.rb:18]: yeah i saw you'd done your test like that i was like damn thats good

File: spec/plane_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require 'plane'
+require 'airport'
+
+describe Plane do
+
+  it 'is flying' do
```
[the remark this answers is not in the document]
Comment (ashtroboy79) [reply spec/plane_spec.rb:6]: mmmm i hadnt thought of that,

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+class Airport
+  DEFAULT_CAPAPCITY = 2
+
+  attr_reader :planes
+  def initialize(capacity = DEFAULT_CAPAPCITY)
+    @weather = Weather.new.stormy?
```
[the remark this answers is not in the document]
Comment (ashtroboy79) [reply lib/airport.rb:6]: it should only generate the weather once, but now im wondering if it messes something up in my tests :/

File: lib/plane.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (edpackard) [reply lib/plane.rb:8]: I think one of your tests was `expect(subject.flying?).to be true` - why not `subject.flying` as the @flying variable will return the true or false status of the plane, while flying? _can change_ that value (and as the code stands, will default to true if no argument is passed - and your test isn't passing a value).

File: spec/weather_spec.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+require 'weather'
+
+describe Weather do
+  
+  it 'can be sunny' do
```
[the remark this answers is not in the document]
Comment (ashtroboy79) [reply spec/weather_spec.rb:5]: thinking about it now, probably not.

File: spec/airport_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (edpackard) [reply spec/airport_spec.rb:9]: It takes some getting used to, but you've set up a good suite of tests, so feeling safe in the knowledge that you can always return to an earlier commit, have a play with doubling plane objects - have a look at the doubling advice on the relevant boris bikes chapters for some good tips

File: lib/airport.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (edpackard) [reply lib/airport.rb:6]: So if the airport is generated with stormy weather, it will never be able to land/take off planes?

File: lib/airport.rb (hunk 1 above)
Comment (edpackard) [reply lib/airport.rb:31]: yeah, or change this one to `check_weather?`
